optimade/grammar/v0.10.0.lark

```diff
@@ -0,0 +1,83 @@
+// optimade v0.10.0 grammar spec in lark grammar format
+// The top-level "filter" rule:
+?start: filter
```

Comment:
Is it necessary to have filter here? Could we not just go directly to `expression*` as we did in 0.9.7 of the grammar, where we also got rid of `filter` - but maybe it's needed, since `start` is now optional?

Reply:
I would keep the `filter` because it is in the spec but I would get rid of the `start`. We can use `start='filter'` kw argument for the [Lark constructor](https://lark-parser.readthedocs.io/en/latest/classes/#classes-reference).

Reply:
BTW, Do we allow empty `' '` input? The current version of grammar (`Filter = [Spaces], Expression ;`) requires at least a single valid expression which cannot be empty.

Reply:
I just realised that probably I misunderstand the question. So `filter` should be just a rule (there is no matching to the "filter word) and the `?start` just means that the `start` rule will be skipped from the `Tree`. Based on the current implementation of the `Transformer`, it looks like that we can use the same '?' trick to simplify the Tree for the `value`, `constant`, `comparison` rules as well.

Reply:
> (...) So `filter` should be just a rule (there is no matching to the "filter word) (...)

This was exactly why I think it was removed. The grammar should be abstract and somewhat agnostic to the keywords.
Since I am not too familiar with lark, I am not sure what the implications of the question mark will pose in general, but that's what I consider the tests to be for :)

Reply:
> I just realised that probably I partly misunderstood the question. So `filter` should be just a rule (there is no matching to the "filter word) and the `?start` just means that the `start` rule will be skipped from the `Tree`. Based on the current implementation of the `Transformer`, it looks like that we can use the same '?' trick for the `value`, `constant`, `comparison` rules as well to simplify the Lark Tree.

How far are you in this "optimization"? Do you need some assistance? Will be happy to try and help and push some changes - although it will change the transformers slightly.